server/controllers/excelController.js

```diff
@@ -205,184 +205,172 @@ excelController.getRelationships = async (req, res, next) => {
+        tempObj[tables[0]] = 'many';
+        tempObj[tables[1]] = 'one';
+        relationships.push(tempObj);
+      } else if (leftTableCount === rightTableCount) {
+        tempObj[tables[0]] = 'one';
+        tempObj[tables[1]] = 'one';
+        relationships.push(tempObj);
       }
-
-      addForeignKey(leftTableName, rightTableName);
     }
 
-    // if relationship is one-one, use the first table as the left table
-    // opportunity for improvement here - get user input on which table is the left table and/or let them update it after the fact and regenerate sql
-    else if (relType.has('one')) {
-      addForeignKey(relTables[0], relTables[1])
-    }
+    // adds foreign key to left table, which points to right table's primary key
+    const addForeignKey = (leftTableName, rightTableName) => {
+      for (let obj of res.locals.output) {
+        if (obj.tableName === leftTableName) {
+          foreignKeyObj = {};
+          foreignKeyObj[(`${rightTableName}_id`)] = {
+            linkedTable: `${rightTableName}._id`,
+            type: 'INT'
+          }
+
+          obj.columns.push(foreignKeyObj);
+        }
+      }
+    };
 
-    // if relationship is many-many, add join table, then add both foreign keys to join table
-    else if (relType.has('many')) {
-      const joinTableName = `${relTables[0]}_${relTables[1]}`;
-      const joinTable = {
-        tableName: joinTableName,
-        columns: [{ "_id": { primaryKey: true, type: 'INT' } }]
-      };
+    for (let relationship of relationships) {
+      const relType = new Set(Object.values(relationship));
+      const relTables = Object.keys(relationship)
+
+      // if relationship is one-many, left table becomes "one" and right table becomes "many"
+      if (relType.has('one') && relType.has('many')) {
+        let leftTableName;
+        let rightTableName;
+        for (let tableName in relationship) {
+          if (relationship[tableName] === 'one') leftTableName = tableName;
+          else rightTableName = tableName;
+        }
 
-      output.push(joinTable);
+        addForeignKey(leftTableName, rightTableName);
+      }
 
-      addForeignKey(joinTableName, relTables[0])
-      addForeignKey(joinTableName, relTables[1])
-    }
-  }
+      // if relationship is one-one, use the first table as the left table
+      // opportunity for improvement here - get user input on which table is the left table and/or let them update it after the fact and regenerate sql
+      else if (relType.has('one')) {
+        addForeignKey(relTables[0], relTables[1])
+      }
 
-  res.json(output)
+      // if relationship is many-many, add join table, then add both foreign keys to join table
+      else if (relType.has('many')) {
+        const joinTableName = `${relTables[0]}_${relTables[1]}`;
+        const joinTable = {
+          tableName: joinTableName,
+          columns: [{ "_id": { primaryKey: true, type: 'INT' } }]
```

Comment:
I am assuming this is the primary key for the `many-many` relationships? If so the type needs to be `SERIAL`